test/integration/smoke/test_deploy_vm_extra_config_data.py

```diff
@@ -0,0 +1,360 @@
+                                   user=self.hostConfig['username'],
+                                   passwd=self.hostConfig['password'])
+            virsh_cmd = 'virsh dumpxml %s' % instance_name
+            xml_res = ssh_client.execute(virsh_cmd)
+            xml_as_str = ''.join(xml_res)
+
+            extraconfig_decoded_xml = '<config>' + urllib.unquote(extraconfig) + '</config>'
+
+            # Root XML Elements
+            parser = etree.XMLParser(remove_blank_text=True)
+            domain_xml_root = ET.fromstring(xml_as_str, parser=parser)
+            decoded_xml_root = ET.fromstring(extraconfig_decoded_xml, parser=parser)
+            for child in decoded_xml_root:
+                find_element_in_domain_xml = domain_xml_root.find(child.tag)
+
+                # Fail if extra config is not found in domain xml
+                self.assertNotEquals(
+                    0,
+                    len(find_element_in_domain_xml),
+                    'Element tag from extra config not added to VM'
+                )
+
+                # Compare found XML node with extra config node
+                is_a_match = self.elements_equal(child, find_element_in_domain_xml)
+                self.assertEquals(
+                    True,
+                    is_a_match,
+                    'The element from tags from extra config do not match with those found in domain xml'
+                )
+        finally:
+            self.destroy_vm(response.id)
+
+    @attr(tags=["devcloud", "advanced", "advancedns", "smoke", "basic", "sg"], required_hardware="true")
+    def test_deploy_vm__with_extraconfig_vmware(self):
+        if self.hypervisor.lower() != 'vmware':
+            raise self.skipTest("Skipping test case for non-vmware hypervisor")
+
+        extraconfig = None
+        response = self.deploy_vm('vmware', extraconfig)
+        host_id = response.hostid
+        host = list_hosts(
+            self.apiclient,
+            id=host_id)
+
+        instance_name = response.instancename
+        host_ipaddress = host[0].ipaddress
+
+        ssh_client = SshClient(host_ipaddress, port=22,
+                               user=self.hostConfig['username'],
+                               passwd=self.hostConfig['password'])
+
+        extraconfig_decoded = urllib.unquote(extraconfig)
+        config_arr = extraconfig_decoded.splitlines()
+
+        for config in config_arr:
+            grep_config = "cat /vmfs/volumes/eaa90eaf-b789d201/%s/%s.vmx | grep -w '%s'" % (
+                instance_name, instance_name, config)
+            result = ssh_client.execute(grep_config)
+            self.assertNotEquals(
+                0,
+                len(result),
+                'Extra  configuration not found in instance vmx file'
+            )
+
+        self.destroy_vm(response.id)
+
+    @attr(tags=["devcloud", "advanced", "advancedns", "smoke", "basic", "sg"], required_hardware="true")
+    def test_deploy_vm__with_extraconfig_xenserver(self):
```

Comment:
this test fails with the following error: 
```
[root@trl-1748-k-M7-bstoyanov-marvin ~]# cat /marvin//MarvinLogs/test_deploy_vm_extra_config_data_KGOSEO/failed_plus_exceptions.txt
2018-12-06 09:11:05,135 - CRITICAL - FAILED: test_deploy_vm__with_extraconfig_xenserver: ['Traceback (most recent call last):\n', '  File "/usr/lib64/python2.7/unittest/case.py", line 369, in run\n    testMethod()\n', '  File "/marvin/tests/smoke/test_deploy_vm_extra_config_data.py", line 343, in test_deploy_vm__with_extraconfig_xenserver\n    \'Extra  configuration not found in VM param list\'\n', '  File "/usr/lib64/python2.7/unittest/case.py", line 553, in assertEqual\n    assertion_func(first, second, msg=msg)\n', '  File "/usr/lib64/python2.7/unittest/case.py", line 546, in _baseAssertEqual\n    raise self.failureException(msg)\n', 'AssertionError: Extra  configuration not found in VM param list\n']

```

Reply:
Fixed and updated test

====Trying SSH Connection: Host:10.2.2.241 User:root                                   Port:22 RetryCnt:60===
===SSH to Host 10.2.2.241 port : 22 SUCCESSFUL===
{Cmd: xe vm-list name-label=i-2-52-VM params=uuid  via Host: 10.2.2.241} {returns: [u'uuid ( RO)    : 7a8c243e-c8d4-1419-2811-7c73fc8ffe7d', u'', u'']}
{Cmd: xe vm-param-get param-name=PV-bootloader uuid=7a8c243e-c8d4-1419-2811-7c73fc8ffe7d via Host: 10.2.2.241} {returns: [u'pygrub']}
{Cmd: xe vm-param-get param-name=PV-args uuid=7a8c243e-c8d4-1419-2811-7c73fc8ffe7d via Host: 10.2.2.241} {returns: [u'hvc0']}
{Cmd: xe vm-param-get param-name=HVM-boot-policy uuid=7a8c243e-c8d4-1419-2811-7c73fc8ffe7d via Host: 10.2.2.241} {returns: [u'']}
=== TestName: test_deploy_vm__with_extraconfig_xenserver | Status : SUCCESS ===